spec/lib/msf/core/exploit/remote/remote/http/http_cookie_jar_spec.rb

```diff
@@ -0,0 +1,278 @@
+      expect(cookie_jar.cookies[0]).to_not eq(c)
+    end
+
+    it 'variable not a subclass of ::HttpCookie will raise TypeError' do
+      int = rand(1..100)
+
+      expect do
+        cookie_jar.add(int)
+      end.to raise_error(TypeError)
+    end
+  end
+
+  describe 'delete' do
+    it 'used on an empty jar will return nil' do
+      # cookie_jar made in before
+
+      n = cookie_jar.delete(cookie)
+
+      expect(n).to eq(nil)
+    end
+
+    it 'passed cookie with same name, domain, and path as cookie in jar, will delete cookie in jar' do
+      c = cookie
+      c_dup = cookie
+      c_dup.name = c.name
+      c_dup.domain = c.domain
+      c_dup.path = c.path
+
+      cookie_jar.add(c)
+      cookie_jar.delete(c_dup)
+
+      expect(cookie_jar.empty?).to eq(true)
+    end
+
+    it 'passed a cookie different name, domain, and path as cookie in jar, will not delete cookie in jar' do
+      c = cookie
+      c_dup = cookie
+      c_dup.name = c.name + random_string(1, 1)
+      c_dup.domain = c.domain + random_string(1, 1)
+      c_dup.path = c.path + random_string(1, 1)
+
+      cookie_jar.add(c)
+      cookie_jar.delete(c_dup)
+
+      expect(cookie_jar.cookies.length).to eq(1)
+      expect(cookie_jar.cookies[0]).to eql(c)
+    end
+
+    it 'variable not a subclass of ::HttpCookie will not raise TypeError when the cookie_jar is empty' do
+      int = rand(1..100)
+
+      n = cookie_jar.delete(int)
+
+      expect(n).to eq(nil)
+      expect(cookie_jar.empty?).to eq(true)
+    end
+
+    it 'variable not a subclass of ::HttpCookie will raise TypeError when the cookie_jar is not empty' do
+      cookie_jar.add(cookie)
+      int = rand(1..100)
+
+      expect do
+        cookie_jar.delete(int)
+      end.to raise_error(TypeError)
+    end
+  end
+
+  describe 'cleanup' do
+    it 'will make no changes to an empty cookiejar' do
+      # empty cookie_jar made in before
+
+      cookie_jar.cleanup
+
+      expect(cookie_jar.empty?).to eq(true)
+    end
+
+    it 'will remove expired cookies with max_age value' do
+      expired_cookies = [cookie, cookie]
+      expired_cookies[0].max_age = 1
+      expired_cookies[1].max_age = 1
+      expired_cookies[0].created_at = Time.local(2008, 9, 5, 10, 5, 1)
+      expired_cookies[1].created_at = Time.local(2008, 9, 5, 10, 5, 1)
+      cookies = [cookie, cookie]
+      cookies[0].max_age = 10000
+      cookies[1].max_age = 10000
+      cookies[0].created_at = Time.now
+      cookies[1].created_at = Time.now
+
+      cookies.map { |c| cookie_jar.add(c) }
+      expired_cookies.map { |c| cookie_jar.add(c) }
+      cookie_jar.cleanup
+
+      expect(cookie_jar.cookies.length).to eq(2)
+      expect(cookie_jar.cookies.include?(expired_cookies[0])).to eq(false)
+      expect(cookie_jar.cookies.include?(expired_cookies[1])).to eq(false)
+      expect(cookie_jar.cookies.include?(cookies[0])).to eq(true)
+      expect(cookie_jar.cookies.include?(cookies[1])).to eq(true)
```

Comment:
I could be wrong, but is this the same as
```
expect(cookie_jar.cookies).to match_array(cookies)
```

Reply:
Your code is the same, but it's less granular and doesn't let you know exactly what the issue is straight away